engines/sword25/package/packagemanager.cpp

```diff
@@ -198,6 +198,21 @@ byte *PackageManager::getFile(const Common::String &fileName, uint *fileSizePtr)
 	int bytesRead = in->read(buffer, in->size());
 	delete in;
 
+	// Modify the buffer to enable internal debugger if needed
+	if (debugChannelSet(-1, kDebugInternalDebugger) && fileName.equals("/system/internal_config.lua")) {
+		const char oldSentence[] = "ENGINE_RELEASE_TYPE = 'pub'";
+		const char newSentence[] = "ENGINE_RELEASE_TYPE = 'dev'";
+		const int sentenceLen = strlen(oldSentence);
+		byte *pos = buffer;
+		while (*pos) {
+			if (memcmp(pos, oldSentence, sentenceLen) == 0) {
+				memcpy(pos, newSentence, sentenceLen);
+				break;
+			}
+			pos++;
+		}
+	}
+
```

Comment:
This can be simplified as:

```suggestion
	if (debugChannelSet(-1, kDebugInternalDebugger) && fileName.equals("/system/internal_config.lua")) {
		char *found = strstr(buffer, "ENGINE_RELEASE_TYPE = 'pub'");
		if (found != nullptr) {
			memcpy(found + 23, "dev", 3);
		}
	}

```

I haven't tested this, so please do :)

Reply:
Thanks! Since buffer's type is `unsigned char*`, we need to use `strstr((char *)buffer, "ENGINE_RELEASE_TYPE = 'pub'");` and it works. Shall I apply this change and commit again?

Reply:
Sure, go ahead